modules/data-streams/src/main/java/org/elasticsearch/datastreams/lifecycle/DataStreamLifecycleService.java

```diff
@@ -310,7 +331,230 @@ void run(ClusterState state) {
+            affectedIndices.addAll(
+                checkEarlyRoundsForMaintenanceOrDownsampleLastRound(dataStream, backingIndex, downsamplingRounds, metadata)
+            );
         }
+
+        return affectedIndices;
+    }
+
+    /**
+     * Iterate over the matching downsampling rounds for the backing index (if any) and either wait for an early round to complete,
+     * add an early completed downsampling round to the data stream, or otherwise trigger the last matching downsampling round.
+     */
+    private Set<Index> checkEarlyRoundsForMaintenanceOrDownsampleLastRound(
+        DataStream dataStream,
+        IndexMetadata backingIndex,
+        List<DataStreamLifecycle.Downsampling.Round> downsamplingRounds,
+        Metadata metadata
+    ) {
+        assert dataStream.getIndices().contains(backingIndex.getIndex())
+            : "the provided backing index must be part of data stream:" + dataStream.getName();
+        assert downsamplingRounds.isEmpty() == false : "the index should be managed and have matching downsampling rounds";
+        Set<Index> affectedIndices = new HashSet<>();
+        DataStreamLifecycle.Downsampling.Round lastRound = downsamplingRounds.get(downsamplingRounds.size() - 1);
+
+        Index index = backingIndex.getIndex();
+        String indexName = index.getName();
+        for (DataStreamLifecycle.Downsampling.Round round : downsamplingRounds) {
+            // the downsample index name for each round is deterministic
+            String downsampleIndexName = DownsampleConfig.generateDownsampleIndexName(
+                DOWNSAMPLED_INDEX_PREFIX,
+                backingIndex,
+                round.config().getFixedInterval()
+            );
+            IndexMetadata targetDownsampleIndexMeta = metadata.index(downsampleIndexName);
+            boolean targetDownsampleIndexExists = targetDownsampleIndexMeta != null;
+
+            if (targetDownsampleIndexExists) {
+                IndexMetadata.DownsampleTaskStatus downsampleStatus = INDEX_DOWNSAMPLE_STATUS.get(targetDownsampleIndexMeta.getSettings());
+                if (downsampleStatus.equals(UNKNOWN) && round.equals(lastRound)) {
+                    // target downsampling index exists and is not a downsampling index (name clash?)
+                    // we fail now but perhaps we should just randomise the name?
+                    String previousError = errorStore.getError(indexName);
+
+                    errorStore.recordError(indexName, new ResourceAlreadyExistsException(targetDownsampleIndexMeta.getIndex().getName()));
+                    // To avoid spamming our logs, we only want to log the error once.
+                    if (previousError == null || previousError.equals(errorStore.getError(indexName)) == false) {
+                        logger.error(
+                            "Data stream lifecycle service is unable to downsample backing index [{}] for data stream [{}] and "
```

Comment:
I'll move this dance into the error store itself in a follow-up (so non-spamming-logging comes by default for all operations)